diffgraph/cli.py

```
@@ -300,22 +315,44 @@ def main(
+                destination = Path(output or "diffgraph.html")
+                try:
+                    html_path = HtmlFormatter(artifact).write(destination)
+                except (OSError, TypeError, ValueError) as error:
+                    raise click.ClickException(
+                        f"could not write {destination}: {error}"
+                    ) from error
+                click.echo(f"✅ HTML report generated: {html_path}")
+                if not no_open:
+                    click.echo("🌐 Opening report in browser...")
+                    if sys.platform == 'darwin':
+                        subprocess.run(['open', str(html_path)])
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)